### PR TITLE
fix abp module bugs

### PR DESCRIPTION
**Pull request by jdorweiler**
Fixing 3 bugs:
- regex filter parsing
- domain anchors with wildcards
- checking whitelist options

Unit tests to go with this are here: https://github.com/duckduckgo/chrome-zeroclickinfo/pull/229

### CODE REVIEW

File: src/abp-filter-parser.js [hunk 1]
```diff
@@ -210,14 +210,7 @@ export function parseFilter(input, parsedFilterData, bloomFilter, exceptionBloom
...
   if (parsedFilterData.isRegex) {
-      
-      // check for escaped forward slashes
-      if (input.match(/\/$/)) {
-          parsedFilterData.data = input.slice(beginIndex + 1);
-      } else {
-          parsedFilterData.data = input.slice(beginIndex + 1, -1);
-      }
-      
+      parsedFilterData.data = input.slice(beginIndex + 1, -1);
```
Comment (laurengarcia) [inline src/abp-filter-parser.js:213]: Question here: the previous code was checking for escaped forward slashes, what about this PR makes that unnecessary now?

File: src/abp-filter-parser.js [hunk 2]
```diff
@@ -480,10 +473,27 @@ export function matchesFilter(parsedFilterData, input, contextParams = {}, cache
...
     }
 
-    return !isThirdPartyHost(parsedFilterData.host, cachedInputData.currentHost) &&
-      indexOfFilter(input, parsedFilterData.data) !== -1;
+    // domain anchored, first check if we're on the correct domain
+    if(!isThirdPartyHost(parsedFilterData.host, cachedInputData.currentHost)) {
+        // check wildcard filters
+        if (parsedFilterData.rawFilter.match(/\*/)) {
+            return wildcardMatch(parsedFilterData, input)
+        // or check normal filtes
```
Comment (laurengarcia) [inline src/abp-filter-parser.js]: Typo ^^

File: src/abp-filter-parser.js [hunk 3]
```diff
@@ -585,7 +594,7 @@ export function matches(parserData, input, contextParams = {}, cachedInputData =
     // Check for exceptions only when there's a match because matches are
     // rare compared to the volume of checks
     let exceptionBloomFilterMiss = parserData.exceptionBloomFilter && !parserData.exceptionBloomFilter.substringExists(cleanedInput, fingerprintSize);
-    if (!exceptionBloomFilterMiss || hasMatchingFilters(parserData.exceptionFilters, parserData, input, contextParams, cachedInputData)) {
+    if (!exceptionBloomFilterMiss && hasMatchingFilters(parserData.exceptionFilters, parserData, input, contextParams, cachedInputData)) {
```
Comment (laurengarcia) [inline src/abp-filter-parser.js:597]: Question: why the switch from `||` to `&&` here?

Comment (laurengarcia) [reply src/abp-filter-parser.js:597]: [EDIT] Just found it in your technical design template, I see there is a link in there to the updated Brave version of this, do you also maybe want to point at this line from that doc as well? Would be helpful as a cross-reference.

File: src/abp-filter-parser.js (hunk 1 above)
Comment (jdorweiler) [reply src/abp-filter-parser.js:213]: That was actually added by me back when we started using the module. I ran into a parsing error and at the time it looked like the escaped forward slash was the problem.
https://github.com/duckduckgo/abp-filter-parser/commit/3173b9e2bd018ad95e6ab73a5c1b6b36792e66a7#diff-f328d5f0285674b0f09e41c58d8e375a

The real problem was that we weren't correctly finding the start of the filter options in filters with multiple '$' characters.

File: src/abp-filter-parser.js (hunk 3 above)
Comment (jdorweiler) [reply src/abp-filter-parser.js:597]: 👍

**Pull request merged**